### PR TITLE
Bug 1113851 - [Settings] Change time format to 24-hour, write gaia-ui te...

### PR DESCRIPTION
**Pull request by bebef1987**
...st

### CODE REVIEW

**Comment by try-server-hook** [comment]
[bebef1987 Florin Strugariu (bebef1987)](https://github.com/bebef1987) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b9715e9e5f04)

File: tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_change_time_format.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+
+        # Set the time on the device
+        _seconds_since_epoch = self.marionette.execute_script("""
+                    var today = new Date();
+                    var yr = today.getFullYear();
+                    var mth = today.getMonth();
+                    var day = today.getDate();
+                    return new Date(yr, mth, day, 22, 0, 0).getTime();""")
```
Comment (geoelectric) [inline tests/python/gaia-ui-tests/gaiatest/tests/functional/settings/test_settings_change_time_format.py]: Re: Robert's note, I don't think you want to count on the controlling system and the target having the same time. So doing all time operations in the context of the target makes sense here.

**Comment by geoelectric** [comment]
So, codewise this looks pretty good.

Couple of concerns.

Biggest concern is that this has the side effect of changing the time on the target device, and doesn't change it back. I think we should save off and offset and restore it.

Setup:

1) read the current time. subtract from epoch of (today 10pm), which could be a positive or negative offset depending on when the test is run.

2) Set to today 10pm, as now.

Test:

Do the test as written.

Teardown:

Reset time to current - offset. 

That should restore things as closely as possible.

Second concern is that nothing is explicitly checking that the Date/Time is starting out at 12 hours. We do implicitly check via the status bar verification of 10:00 to start, so this is a more minor concern. If that error gets thrown, though, not sure it'll be obvious to someone triaging why it happened. 

It'd be more robust to save off the current format setting and set to 12 hour in setup, then run the test as written, then restore the original setting in teardown.

Final concern isn't an automation problem, it's a test problem. Nobody's checking the field on Date/Time that has the current time in it. That should also flip to 24 hours. It might be worth adding the check here, even though it's over and above the MozTrap test as written.

I'd like to hear back on at least the first concern before reviewing.

**Comment by bebef1987** [comment]
make sure in the setup of each test that the ime and date of the phone are accurate. No need for any teardown steps here

**Comment by chirarobert** [comment]
The test failed on treeherder. @bebef1987

**Comment by try-server-hook** [comment]
[bebef1987 Florin Strugariu (bebef1987)](https://github.com/bebef1987) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2c9f243cdc5f)

**Comment by try-server-hook** [comment]
[chirarobert chirarobert](https://github.com/chirarobert) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4476d42dfe84)

**Pull request merged**